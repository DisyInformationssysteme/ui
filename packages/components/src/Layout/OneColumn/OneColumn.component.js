import PropTypes from 'prop-types';
import classnames from 'classnames';
import omit from 'lodash/omit';

import theme from './OneColumn.module.scss';
import { FloatingDrawer, Tabs } from '@talend/design-system';

/**
 * @param {object} props react props
 * @example
 <OneColumn name="Hello world"></OneColumn>
 */
function OneColumn({ drawers, children, tabs, ...props }) {
	const container = classnames('tc-layout-one-column', theme.main);
	const style = {
		overflow: 'auto',
		height: '100%',
		display: 'flex',
		flexDirection: 'column',
	};
	return (
		<div
			role="main"
			id="tc-layout-main"
			tabIndex="-1"
			className={container}
			{...omit(props, 'getComponent')}
		>
			<FloatingDrawer.Container>
				{tabs && <Tabs {...tabs} />}
				<div style={style}>{children}</div>
				{drawers.map((drawer, index) => (
					<FloatingDrawer key={index} visible>
						{drawer}
					</FloatingDrawer>
				))}
			</FloatingDrawer.Container>
		</div>
	);
}

OneColumn.displayName = 'OneColumn';

OneColumn.propTypes = {
	children: PropTypes.node,
	drawers: PropTypes.arrayOf(PropTypes.node),
	tabs: PropTypes.shape(TabBar.propTypes),
};

export default OneColumn;
