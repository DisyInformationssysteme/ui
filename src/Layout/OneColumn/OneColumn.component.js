import React from 'react';

/**
 * @param {object} props react props
 * @example
<OneColumn name="Hello world"></OneColumn>
 */
function OneColumn(props) {
	return (<div>{props.one}</div>);
}

OneColumn.propTypes = {
	one: React.PropTypes.element,
};

export default OneColumn;
