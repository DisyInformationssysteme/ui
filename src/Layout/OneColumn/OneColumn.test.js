import React from 'react';
import renderer from 'react-test-renderer';

import OneColumn from './OneColumn.component';

describe('OneColumn', () => {
	it('should render its name', () => {
		const wrapper = renderer.create(<OneColumn name="Hello world" />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
