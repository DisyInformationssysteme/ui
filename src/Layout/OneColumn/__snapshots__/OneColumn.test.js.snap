exports[`OneColumn should render its name 1`] = `<div />`;
